File: app/src/main/java/com/example/khughes/machewidget/db/UserInfoDao.java
package com.example.khughes.machewidget.db;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.khughes.machewidget.UserInfo;

import java.util.List;

@Dao
public interface UserInfoDao {
    @Insert
    void insertUserInfo(UserInfo info);

    @Query("SELECT * FROM user_info WHERE userId LIKE :userId")
    UserInfo findUserInfo(String userId);

    @Query("UPDATE user_info SET username = '', password = ''")
    void clearCredentials();

    @Delete
    void deleteUserInfo(UserInfo user);

    @Query("DELETE FROM user_info WHERE userId LIKE :userId")
    void deleteUserInfoByUserId(String userId);

    @Query("UPDATE user_info SET programState = :state WHERE userId = :userId")
    void updateProgramState(String state, String userId);

    @Query("UPDATE user_info SET lastModified = :mod WHERE userId = :userId")
    void updateLastModified(String mod, String userId);

    @Update
    void updateUserInfo(UserInfo info);
}

File: app/src/main/java/com/example/khughes/machewidget/db/UserInfoDao.kt
package com.example.khughes.machewidget.db

import androidx.room.*
import com.example.khughes.machewidget.UserInfo

@Dao
interface UserInfoDao {
    @Insert
    fun insertUserInfo(info: UserInfo)

    @Query("SELECT * FROM user_info")
    fun findUserInfo(): List<UserInfo>

    @Query("SELECT * FROM user_info WHERE userId LIKE :userId")
    fun findUserInfo(userId: String): UserInfo?

    @Delete
    fun deleteUserInfo(user: UserInfo)

    @Query("DELETE FROM user_info WHERE userId LIKE :userId")
    fun deleteUserInfoByUserId(userId: String)

    @Query("DELETE FROM user_info")
    fun deleteAllUserInfo()

    @Query("UPDATE user_info SET programState = :state WHERE userId = :userId")
    fun updateProgramState(state: String, userId: String)

    @Query("UPDATE user_info SET lastModified = :mod WHERE userId = :userId")
    fun updateLastModified(mod: String, userId: String)

    @Update
    fun updateUserInfo(info: UserInfo)
}